app/controllers/preview_controller.rb

```diff
@@ -2,24 +2,26 @@
 
 class PreviewController < ApplicationController
   def create
-    Document.transaction do
-      @document = Document.with_current_edition.lock.find_by_param(params[:id])
+    Edition.find_and_lock_current(document: params[:document]) do |edition|
```

Comment:
I'm wondering if we really need a lock here?

Reply:
We do as we modify the edition model, we set revision_synced. We also potentially modify some asset models.

Reply:
Oh yes - I forgot about all the coordination the preview service does!